### PR TITLE
I18n/rbac settings plugins

### PR DESCRIPTION
**Pull request by soupette**
### What does it do?

It refactor the code of the plugins and settings tabs

### CODE REVIEW

File: packages/strapi-admin/admin/src/components/Roles/Permissions/PluginsAndSettingsPermissions/PermissionRow/index.js [hunk 1]
```diff
@@ -23,6 +23,8 @@ const PermissionRow = ({
       : permissions.category;
   }, [permissions]);
 
+  console.log({ permissions, categoryName });
```
Comment (mfrachet) [inline packages/strapi-admin/admin/src/components/Roles/Permissions/PluginsAndSettingsPermissions/PermissionRow/index.js:26]: You can strip off this, I think 😛

File: packages/strapi-admin/ee/admin/containers/Roles/CreatePage/Permissions/utils/tests/createDefaultPluginsFormFromLayout.test.js [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+  },
+  {
+    id: 'admin::has-same-role-as-creator',
+    displayName: 'Has same role as creator',
+    category: 'default',
+  },
+];
+
+describe('ADMIN | COMPONENTS | Permissions | utils', () => {
+  describe('createSubCategoryForm', () => {
```
Comment (mfrachet) [inline packages/strapi-admin/ee/admin/containers/Roles/CreatePage/Permissions/utils/tests/createDefaultPluginsFormFromLayout.test.js:20]: Can you write a test with at least one leaf true?

File: packages/strapi-admin/ee/admin/containers/Roles/CreatePage/Permissions/utils/tests/createDefaultPluginsFormFromLayout.test.js [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+            'admin::has-same-role-as-creator': false,
+          },
+        },
+      };
+
+      expect(createSubCategoryForm(actions, conditions)).toEqual(expected);
+    });
+  });
+
+  describe('createChildrenDefaultForm', () => {
```
Comment (mfrachet) [inline packages/strapi-admin/ee/admin/containers/Roles/CreatePage/Permissions/utils/tests/createDefaultPluginsFormFromLayout.test.js:58]: Can you write a test with at least one leaf true?

File: packages/strapi-admin/ee/admin/containers/Roles/CreatePage/PluginsAndSettings/Row/index.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
+import React, { useMemo } from 'react';
+import { Flex, Text } from '@buffetjs/core';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { PermissionsWrapper, RowContainer } from 'strapi-helper-plugin';
+import PropTypes from 'prop-types';
+import { useIntl } from 'react-intl';
+// import SubCategory from 'ee_else_ce/components/Roles/Permissions/PluginsAndSettingsPermissions/PermissionRow/SubCategory';
```
Comment (mfrachet) [inline packages/strapi-admin/ee/admin/containers/Roles/CreatePage/PluginsAndSettings/Row/index.js:7]: I think we can remove this one too

File: packages/strapi-admin/ee/admin/containers/Roles/CreatePage/PluginsAndSettings/Row/index.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  const { formatMessage } = useIntl();
+
+  const handleClick = () => {
+    onOpenCategory(name);
+  };
+
+  const categoryName = useMemo(() => {
+    const split = name.split('::');
+
+    return split[split.length - 1];
```
Comment (mfrachet) [inline packages/strapi-admin/ee/admin/containers/Roles/CreatePage/PluginsAndSettings/Row/index.js:29]: Nitpicky: when you don't really care about mutating the array, you can simply rely on `split.pop()` 😊

**Review by mfrachet — APPROVED** [review]
Thanks to the documentation you've written and the according tests, it was waaaay easier to read 😊 .

Thank you for this 👍🏻 

I'm having some tiny comments about uncommented lines, tiny suggestions. approving for me 👍🏻

**Review by HichamELBSI — APPROVED** [review]
LGTM

File: packages/strapi-admin/admin/src/components/Roles/Permissions/PluginsAndSettingsPermissions/PermissionRow/index.js (hunk 1 above)
Comment (soupette) [reply packages/strapi-admin/admin/src/components/Roles/Permissions/PluginsAndSettingsPermissions/PermissionRow/index.js:26]: Fixed in #9420

File: packages/strapi-admin/ee/admin/containers/Roles/CreatePage/Permissions/utils/tests/createDefaultPluginsFormFromLayout.test.js (hunk 2 above)
Comment (soupette) [reply packages/strapi-admin/ee/admin/containers/Roles/CreatePage/Permissions/utils/tests/createDefaultPluginsFormFromLayout.test.js:20]: Of course, I'll update the tests in #9420

File: packages/strapi-admin/ee/admin/containers/Roles/CreatePage/PluginsAndSettings/Row/index.js (hunk 4 above)
Comment (soupette) [reply packages/strapi-admin/ee/admin/containers/Roles/CreatePage/PluginsAndSettings/Row/index.js:7]: Fixed in #9420

File: packages/strapi-admin/ee/admin/containers/Roles/CreatePage/PluginsAndSettings/Row/index.js (hunk 5 above)
Comment (soupette) [reply packages/strapi-admin/ee/admin/containers/Roles/CreatePage/PluginsAndSettings/Row/index.js:29]: fixed in #9420

**Pull request merged**